diff -up groff-1.22.4/src/roff/nroff/nroff.sh.1~ groff-1.22.4/src/roff/nroff/nroff.sh
--- groff-1.22.4/src/roff/nroff/nroff.sh.1~	2018-11-18 23:28:27.000000000 +0100
+++ groff-1.22.4/src/roff/nroff/nroff.sh	2018-12-27 14:37:20.812159924 +0100
@@ -123,6 +123,32 @@ case $T in
     T=-T$Tloc ;;
 esac
 
+# copy the man page to a temp file
+TMPFILE=$(mktemp /tmp/man.XXXXXX)
+trap "rm -f $TMPFILE" 0 1 2 3 15
+cat ${1+"$@"} >| ${TMPFILE}
+
+# test the charset encoding of the man page
+if iconv -f utf-8 -t utf-8 -o /dev/null ${TMPFILE} 2>/dev/null
+then
+  charset_in=utf-8
+else
+  # non utf-8 encoding
+  case "${LANGUAGE-${LC_CTYPE-${LANG}}}" in
+    cs*|hr*|hu*|pl*|ro*|sk*|sl*) charset_in=iso-8859-2 ;;
+    el*) charset_in=iso-8859-7 ;;
+    tr*) charset_in=iso-8859-9 ;;
+    bg*) charset_in=cp1251 ;;
+    ja*) charset_in=euc-jp ;;
+    ko*) charset_in=euc-kr ;;
+    ru*) charset_in=koi8-r ;;
+    uk*) charset_in=koi8-u ;;
+    zh_TW*|zh_HK*) charset_in=big5 ;;
+    zh_CN*|zh*) charset_in=gb2312 ;;
+    *) charset_in=iso-8859-1 ;;
+  esac
+fi
+
 # Set up the 'GROFF_BIN_PATH' variable
 # to be exported in the current 'GROFF_RUNTIME' environment.
 
@@ -131,6 +157,6 @@ export GROFF_BIN_PATH
 
 # Load nroff-style character definitions too.
 
-PATH="$GROFF_RUNTIME$PATH" groff -mtty-char $T $opts ${1+"$@"}
+PATH="$GROFF_RUNTIME$PATH" cat $TMPFILE | groff -K $charset_in -mtty-char $T $opts 2>/dev/null
 
 # eof
