diff -up groff-1.23.0/src/roff/nroff/nroff.sh.1~ groff-1.23.0/src/roff/nroff/nroff.sh
--- groff-1.23.0/src/roff/nroff/nroff.sh.1~	2023-07-08 11:20:22.646988062 +0200
+++ groff-1.23.0/src/roff/nroff/nroff.sh	2023-07-08 11:25:06.015958731 +0200
@@ -158,6 +158,32 @@ fi
 # Load nroff-style character definitions too.
 opts="-mtty-char$opts"
 
+# copy the man page to a temp file
+TMPFILE=$(mktemp /tmp/man.XXXXXX)
+trap "rm -f $TMPFILE" 0 1 2 3 15
+cat ${1+"$@"} >| ${TMPFILE}
+
+# test the charset encoding of the man page
+if iconv -f utf-8 -t utf-8 -o /dev/null ${TMPFILE} 2>/dev/null
+then
+  charset_in=utf-8
+else
+  # non utf-8 encoding
+  case "${LANGUAGE-${LC_CTYPE-${LANG}}}" in
+    cs*|hr*|hu*|pl*|ro*|sk*|sl*) charset_in=iso-8859-2 ;;
+    el*) charset_in=iso-8859-7 ;;
+    tr*) charset_in=iso-8859-9 ;;
+    bg*) charset_in=cp1251 ;;
+    ja*) charset_in=euc-jp ;;
+    ko*) charset_in=euc-kr ;;
+    ru*) charset_in=koi8-r ;;
+    uk*) charset_in=koi8-u ;;
+    zh_TW*|zh_HK*) charset_in=big5 ;;
+    zh_CN*|zh*) charset_in=gb2312 ;;
+    *) charset_in=iso-8859-1 ;;
+  esac
+fi
+
 # Set up the 'GROFF_BIN_PATH' variable to be exported in the current
 # 'GROFF_RUNTIME' environment.
 @GROFF_BIN_PATH_SETUP@
@@ -195,7 +221,7 @@ if [ -n "$dry_run" ]
 then
   echo PATH="$GROFF_RUNTIME$PATH" $groff $T $opts ${1+"$@"}
 else
-  PATH="$GROFF_RUNTIME$PATH" $groff $T $opts ${1+"$@"}
+  PATH="$GROFF_RUNTIME$PATH" cat $TMPFILE | $groff -K $charset_in $T $opts ${1+"$@"}
 fi
 
 # Local Variables:
