diff -up groff-1.24.1/lib/stdlib.in.h.2~ groff-1.24.1/lib/stdlib.in.h
diff -up groff-1.24.1/lib/string.in.h.2~ groff-1.24.1/lib/string.in.h
diff -up groff-1.24.1/src/include/config.hin.2~ groff-1.24.1/src/include/config.hin
--- groff-1.24.1/src/include/config.hin.2~	2026-03-14 08:53:45.000000000 +0100
+++ groff-1.24.1/src/include/config.hin	2026-03-15 18:25:08.050863142 +0100
@@ -1886,7 +1886,7 @@
    can be freed via 'free'; it can be used only after declaring 'free'.  */
 /* Applies to: functions.  Cannot be used on inline functions.  */
 #ifndef _GL_ATTRIBUTE_DEALLOC
-# if _GL_GNUC_PREREQ (11, 0)
+# if _GL_GNUC_PREREQ (11, 0) && !defined(__clang__)
 #  define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 # else
 #  define _GL_ATTRIBUTE_DEALLOC(f, i)
